buildrunner/validation/step.py

```diff
@@ -0,0 +1,106 @@
+"""
+Copyright 2023 Adobe
+All Rights Reserved.
+
+NOTICE: Adobe permits you to use, modify, and distribute this file in accordance
+with the terms of the Adobe license agreement accompanying it.
+"""
+
+from typing import Any, Dict, List, Optional, Union
+
+# pylint: disable=no-name-in-module
+from pydantic import BaseModel, Field
+
+
+class StepPypiPush(BaseModel):
+    """ Step pypi push model"""
+    repository: str
+    username: str
+    password: str
+
+
+class Artifact(BaseModel):
+    """ Artifact model """
+    # Intentionally loose restrictions
+    format: Optional[str]
+    type: Optional[Any]
+    compression: Optional[str]
+    push: Optional[bool]
+
+
+class StepRun(BaseModel, extra='forbid'):
+    """ Run model within a step """
+
+    xfail: Optional[bool]
+    services: Optional[Dict[str, Any]]
+    image: Optional[str]
+    cmd: Optional[str]
+    cmds: Optional[List[str]]
+    # Intentionally loose restrictions
+    provisioners: Optional[Dict[str, str]]
+    shell: Optional[str]
+    cwd: Optional[str]
+    user: Optional[str]
+    hostname: Optional[str]
+    dns: Optional[List[str]]
+    dns_search: Optional[str]
+    extra_hosts: Optional[Dict[str, str]]
+    env: Optional[Dict[str, str]]
+    files: Optional[Dict[str, str]]
+    caches: Optional[Dict[str, Union[str, List[str]]]]
+    ports: Optional[Dict[str, str]]
+    volumes_from: Optional[List[str]]
+    ssh_keys: Optional[List[str]] = Field(alias='ssh-keys')
+    artifacts: Optional[Dict[str, Union[Artifact, None]]]
```

Comment:
Isn't this `Optional[Dict[str, Optional[Artifact]]]`? i.e. Optional instead of the Union with None

Reply:
Yes. I like that. It is more concise. I'll make the update.